### PR TITLE
Replace system.network.state with network.connection.state

### PR DESCRIPTION
**Pull request by ChrsMark**
Fixes https://github.com/open-telemetry/semantic-conventions/issues/308

## Changes

ref: https://github.com/open-telemetry/semantic-conventions/issues/308#issuecomment-2504357654

## Merge requirement checklist

* [x] [CONTRIBUTING.md](https://github.com/open-telemetry/semantic-conventions/blob/main/CONTRIBUTING.md) guidelines followed.
* [x] Change log entry added, according to the guidelines in [When to add a changelog entry](https://github.com/open-telemetry/semantic-conventions/blob/main/CONTRIBUTING.md#when-to-add-a-changelog-entry).
  * If your PR does not need a change log, start the PR title with `[chore]`
* [x] [schema-next.yaml](https://github.com/open-telemetry/semantic-conventions/blob/main/schema-next.yaml) updated with changes to existing conventions.

### CODE REVIEW

**Review by mx-psi — APPROVED** [review]

File: model/network/registry.yaml
Comment (AlexanderWert) [inline model/network/registry.yaml:279]: can we move that statement into `note` and in `brief` describe what that attribute stands for?

File: model/system/registry.yaml
Comment (AlexanderWert) [inline model/system/registry.yaml:155]: Since this attribute is being deprecated here, it should be moved to this file: `/model/system/deprecated/registry-deprecated.yaml`

**Review by braydonk — APPROVED** [review]

**Review by dmitryax — APPROVED** [review]

**Comment by ChrsMark** [comment]
@AlexanderWert @open-telemetry/specs-semconv-maintainers that should be good to go?

File: schema-next.yaml
Comment (lmolkova) [inline schema-next.yaml:5]: since the attribute is renamed and the old one is deprecated, we should apply this change to all signals, not just specific metric

File: model/network/registry.yaml
Comment (lmolkova) [inline model/network/registry.yaml:242]: I think this is coming from https://datatracker.ietf.org/doc/html/rfc9293, can we 
- review values and adjust them to match the RFC? E.g. this one should be `closed`
- add link to the RFC in the brief?

File: model/network/registry.yaml
Comment (lmolkova) [inline model/network/registry.yaml:269]: RFC uses `SYN-RECEIVED`

File: model/network/registry.yaml
Comment (lmolkova) [inline model/network/registry.yaml:251]: I don't think there is a delete state in the RFC - do we need to add it?

File: model/network/registry.yaml
Comment (lmolkova) [inline model/network/registry.yaml:280]: I suggest to remove this note. Telemetry describing stateless protocols won't use this attribute and it seems obvious that stateless thing doesn't need a state.

**Review by lmolkova — COMMENTED** [review]
Looks good, left a few suggestions to clean up old enum members

File: model/network/registry.yaml
Comment (ChrsMark) [reply model/network/registry.yaml:242]: I did some digging, looking into where those are used/defined. It seems that they are defined in `hostmetrics` receiver:
1. https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/6f3ff4a37efe4591f728250d4ea9e4ad6672fb2b/receiver/hostmetricsreceiver/internal/scraper/networkscraper/network_others.go#L9-L20 
2. https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/c4bee406f5db233a2bc7f0c69185764fea9fb033/receiver/hostmetricsreceiver/internal/scraper/networkscraper/network_linux.go#L18-L29

Used at https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/c4bee406f5db233a2bc7f0c69185764fea9fb033/receiver/hostmetricsreceiver/internal/scraper/networkscraper/network_scraper.go#L178-L188

These match with what `gopsutil` provides:
1. https://github.com/shirou/gopsutil/blob/62a52f910bd9ad7d48a7416f5c2899386d50cb37/net/net_linux.go#L308
2. https://github.com/shirou/gopsutil/blob/62a52f910bd9ad7d48a7416f5c2899386d50cb37/net/net_windows.go#L565

Moreover, these seem to link to the OS specific definitions:
1. https://github.com/torvalds/linux/blob/cdd30ebb1b9f36159d66f088b61aee264e649d7a/include/net/tcp_states.h#L19
2. https://learn.microsoft.com/en-us/windows/win32/api/tcpmib/ns-tcpmib-mib_tcprow2?redirectedfrom=MSDN#members

So it seems that it is `CLOSED` for Windows while `CLOSE` for Linux. Should we allow both or should we fix the receiver's implementation to correctly map to RFC's values?

@braydonk @dmitryax what are your thoughts since you are the code-owners of the receiver?

File: model/network/registry.yaml
Comment (ChrsMark) [reply model/network/registry.yaml:269]: Same as https://github.com/open-telemetry/semantic-conventions/pull/1632#discussion_r1867330961. 

It is `SYN_RECV` for Linux while `SYN_RECEIVED` for Windows.

File: model/network/registry.yaml
Comment (lmolkova) [reply model/network/registry.yaml:242]: I defer to system semconv group to discuss and decide. I have a preference to stick to RFC definition (which is cross-platform) and allow OS-specific values. In case of `close`/`closed`, the method that translates linux enum number into otel string value can take care of reporting the standard RFC state name.

File: model/network/registry.yaml
Comment (lmolkova) [reply model/network/registry.yaml:269]: replied there, my suggestion is to stick to RFC and it should be easy to translate enum const to otel attribute value string anyway.

File: model/network/registry.yaml
Comment (lmolkova) [reply model/network/registry.yaml:251]: can we postpone providing it? It's not in the RFC and it's always easy to extend enum in semconv.

File: model/network/registry.yaml
Comment (mx-psi) [reply model/network/registry.yaml:280]: Makes sense to me to remove it

File: model/network/registry.yaml
Comment (ChrsMark) [reply model/network/registry.yaml:242]: Discussed in System Metrics WG today. We agree on having the SemConv following what the (cross-platform) RFC defines and then the receiver to properly do the mappings accordingly for each platform/implementation.

I will update this PR accordingly.

File: model/network/registry.yaml
Comment (ChrsMark) [reply model/network/registry.yaml:280]: Removed.

File: model/network/registry.yaml
Comment (ChrsMark) [reply model/network/registry.yaml]: Removed that.

File: model/network/registry.yaml
Comment (ChrsMark) [reply model/network/registry.yaml]: Tuned the PR according to the rfc9293.

**Comment by ChrsMark** [comment]
@lmolkova @open-telemetry/semconv-system-approvers I have tuned the PR so as the state values to be aligned with [rfc9293](https://datatracker.ietf.org/doc/html/rfc9293#section-3.3.2).

I'm not sure if we should explicitly deprecate the enum values that were changed or removed entirely.

**Review by mx-psi — APPROVED** [review]

**Review by lmolkova — APPROVED** [review]
LGTM and thank you!

**Comment by lmolkova** [comment]
tagging @open-telemetry/semconv-security-approvers in case they want to take a look

File: schema-next.yaml
Comment (trisch-me) [inline schema-next.yaml:5]: should here be also noted changes to the metrics?

File: model/system/deprecated/registry-deprecated.yaml
Comment (trisch-me) [inline model/system/deprecated/registry-deprecated.yaml:95]:
```suggestion
        brief: "Deprecated, use `network.connection.state` instead."
```

File: schema-next.yaml
Comment (ChrsMark) [reply schema-next.yaml:5]: Initially it was applied only to metrics but I made it to apply to all as per https://github.com/open-telemetry/semantic-conventions/pull/1632#discussion_r1867013714

**Review by trisch-me — APPROVED** [review]

**Comment by trisch-me** [comment]
@ChrsMark I think one test is failing, otherwise good to go!

**Pull request merged**
